source/shared/cpp/AdaptiveCardsSharedModel/AdaptiveCardsSharedModelUnitTest/AdaptiveCardsSharedModelUnitTest.cpp

```diff
@@ -0,0 +1,206 @@
+#include "stdafx.h"
+#include "CppUnitTest.h"
+#include "TextBlock.h"
+#include <time.h>
+
+using namespace Microsoft::VisualStudio::CppUnitTestFramework;
+using namespace AdaptiveCards;
+using namespace std;
+
+namespace AdaptiveCardsSharedModelUnitTest
+{        
+    TEST_CLASS(DateAndTimeTest)
+    {
+    public:
+        TEST_METHOD(ISO8601TransformTest)
+        {
+            TextBlock blck;
+            blck.SetText("{{DATE(2017-02-13T20:46:30Z, SHORT)}}");
+            Assert::AreEqual<string>("02/13/17",  blck.GetText());
+        }
+        TEST_METHOD(ISO8601TransformTestLONG)
+        {
+            TextBlock blck;
+            string testString = "{{DATE(2017-02-13T20:46:30Z, LONG)}}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("Monday, February 13, 2017", blck.GetText());
+        }
+        TEST_METHOD(ISO8601TransformTestSmallCaseLONG)
+        {
+            TextBlock blck;
+            string testString = "{{DATE(2017-02-13T20:46:30Z, Long)}}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("Monday, February 13, 2017", blck.GetText());
+        }
+        TEST_METHOD(ISO8601TransformTestDefault)
+        {
+            TextBlock blck;
+            string testString = "{{DATE(2017-02-13T20:46:30Z)}}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("02/13/17", blck.GetText());
+        }
+
+        TEST_METHOD(ISO8601TransformBadInputTest)
+        {
+            TextBlock blck;
+            string testString = "{a{DATE(2017-02-13T20:46:30Z, SHORT)}}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("{a{DATE(2017-02-13T20:46:30Z, SHORT)}}", blck.GetText());
+        }
+
+        TEST_METHOD(ISO8601TransformBadInputTestMissingClosingCurlyBracket)
+        {
+            TextBlock blck;
+            string testString = "{{DATE(2017-02-13T20:46:30Z, SHORT)}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("{{DATE(2017-02-13T20:46:30Z, SHORT)}", blck.GetText());
+        }
+
+        TEST_METHOD(ISO8601TransformBadInputTestBadYear)
+        {
+            TextBlock blck;
+            string testString = "{{DATE(2017a02-13T20:46:30Z, SHORT)}}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("{{DATE(2017a02-13T20:46:30Z, SHORT)}}", blck.GetText());
+        }
+
+        TEST_METHOD(TransformToDateTest1)
+        {
+            TextBlock blck;
+            string testString = "{{DATE(2017-02-13T20:46:30Z, SHORT)}}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("02/13/17", blck.GetText());
+        }
+
+        TEST_METHOD(TransformToDateTest2)
```

Comment:
>TransformToDateTest2 [](start = 20, length = 20)

[nit] this is testing TIME, not DATE. Name should probably change

Reply:
[nit] would you mind organizing these tests? TransformToTimeTest2 is placed before TransformToTimeTest1. Optionally, give them more meaningful names, and then the order wouldn't matter as much

---
In reply to: [149153835](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r149153835) [](ancestors = 149153835)

Reply:
will do

---
In reply to: [149176496](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r149176496) [](ancestors = 149176496,149153835)